pkg/types/nutanix/helpers.go

```diff
@@ -37,6 +38,13 @@ const (
 	CategoryValueShared = "shared"
 )
 
+var (
+	// PrismAPICallTimeoutDuration is the timeout for the prism-api calls. It is initialized based on
+	// the DefaultPrismAPICallTimeout value. If the installConfig nutanix platform's PrismAPICallTimeout
+	// is configured, it will set based on that value.
+	PrismAPICallTimeoutDuration = DefaultPrismAPICallTimeout * time.Minute
+)
```

Comment:
Is it possible to avoid this package-level variable? It's a somewhat confusing way to save state.

I would suggest making it a function on the installconfig/platform

Reply:
The package variable is still needed because some use cases of the prism-api call cannot pass in the nutanix.platform as argument, such as at cluster cleanup stage to delete Nutanix resources (VMs, images, etc.).

Reply:
> such as at cluster cleanup stage to delete Nutanix resources (VMs, images, etc.).

It's not clear to me that the current implementation will work to set the timeout during `cluster destroy`. I am skeptical because I don't think the install config asset would be loaded, and therefore the default package variable value would be used. I could be wrong though--it is difficult to reason through. Have you tested it?

Reply:
@patrickdillon fixed. can you take another look?  
Now the prismAPICallTimeoutDuration variable is initialized based on DefaultPrismAPICallTimeout (10 minutes). When the installer is called for cluster destroy, the default timeout (10 minutes) is used. And before the installer make the prism-api call to load image data, the func SetPrismAPICallTimeoutDuration() is called to set the timeout based that value in the install-config if it is configured, otherwise the default timeout value (10 minutes) is used. I did manual test for both cluster creation and cluster destroy. It seems working as expected.